test/framework/cloudstack.go

```diff
@@ -0,0 +1,145 @@
+package framework
+
+import (
+	"os"
+	"testing"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+)
+
+const (
+	cloudstackDomainVar                = "T_CLOUDSTACK_DOMAIN"
+	cloudstackZoneVar                  = "T_CLOUDSTACK_ZONE"
+	cloudstackAccountVar               = "T_CLOUDSTACK_ACCOUNT"
+	cloudstackNetworkVar               = "T_CLOUDSTACK_NETWORK"
+	cloudstackManagementServerVar      = "T_CLOUDSTACK_MANAGEMENT_SERVER"
+	cloudstackSshAuthorizedKeyVar      = "T_CLOUDSTACK_SSH_AUTHORIZED_KEY"
+	cloudstackTemplateRedhat120Var     = "T_CLOUDSTACK_TEMPLATE_REDHAT_1_20"
+	cloudstackTemplateRedhat121Var     = "T_CLOUDSTACK_TEMPLATE_REDHAT_1_21"
+	cloudstackComputeOfferingLargeVar  = "T_CLOUDSTACK_COMPUTE_OFFERING_LARGE"
+	cloudstackComputeOfferingLargerVar = "T_CLOUDSTACK_COMPUTE_OFFERING_LARGER"
+	cloudstackK8sControlPlaneHostVar   = "T_CLOUDSTACK_CONTROL_PLANE_HOST"
+	podCidrVar                         = "T_CLOUDSTACK_POD_CIDR"
+	serviceCidrVar                     = "T_CLOUDSTACK_SERVICE_CIDR"
+)
+
+var requiredCloudStackEnvVars = []string{
+	cloudstackDomainVar,
+	cloudstackZoneVar,
+	cloudstackAccountVar,
+	cloudstackNetworkVar,
+	cloudstackManagementServerVar,
+	cloudstackSshAuthorizedKeyVar,
+	cloudstackTemplateRedhat120Var,
+	cloudstackTemplateRedhat121Var,
+	cloudstackComputeOfferingLargeVar,
+	cloudstackComputeOfferingLargerVar,
+	cloudstackK8sControlPlaneHostVar,
+	podCidrVar,
+	serviceCidrVar,
+}
+
+type CloudStack struct {
+	t           *testing.T
+	fillers     []api.CloudStackFiller
+	podCidr     string
+	serviceCidr string
+}
+
+type CloudStackOpt func(*CloudStack)
+
+func UpdateRedhatTemplate120Var() api.CloudStackFiller {
```

Comment:
We only support redhat for cloudstack for now?

Reply:
I believe those are the only images we have built. I know CAPC itself works with some Ubuntu images as well but I'm not sure if we've tested those at the EKS-A level. Maybe @vignesh-goutham has some more thoughts here